pandora/tagger.py

```diff
@@ -278,6 +278,12 @@ def setup_to_train(self, train_data=None, dev_data=None, test_data=None,
         if self.include_test:
             self.test_contexts = self.pretrainer.transform(
                 tokens=self.test_tokens)
+        try:
+            del train_transformed
+            del dev_transformed
+            del test_transformed
+        except:
+            pass
```

Comment:
Could you actually do a try for each del rather than this way ? Or make sure that vars are initiated to None ?
Because this memory improvement in the potential case where dev do not exist but test does : 


```python
>>> x, y, z = 1, 2, 3
>>> print(y)
2
>>> try:
...   del x
...   del v
...   del y
... except:
...   pass
... 
>>> x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'x' is not defined
>>> y
2

```